src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.cs

```diff
@@ -250,47 +263,62 @@ public void Initialize(Solution solution)
                 // transaction. While some writes can be lost, they are never reordered, and higher layers will recover from that.
                 connection.ExecuteCommand("pragma synchronous=normal", throwOnError: false);
 
-                // First, create all our tables
-                connection.ExecuteCommand(
-$@"create table if not exists ""{StringInfoTableName}"" (
+                CreateTables(connection, MainDBName);
+
+                // Also get the known set of string-to-id mappings we already have in the DB.
+                // Do this in one batch if possible.
+                var fetched = TryFetchStringTableFromMainDB(connection);
+
+                // If we weren't able to retrieve the entire string table in one batch,
+                // attempt to retrieve it for each 
+                var fetchStringTable = !fetched;
+
+                // Try to bulk populate all the IDs we'll need for strings/projects/documents.
+                // Bulk population is much faster than trying to do everything individually.
+                BulkPopulateIds(connection, solution, fetchStringTable);
```

Comment:
these lines are the same from before.  all that happened was i extracted out the table-initialization code since we want to initialize the main tables *and* initialize the tables in the write-cache.